### PR TITLE
Refactor Add initial transition to GetInitialTransition method

### PR DESCRIPTION
**Pull request by sulmar**
It's my proposal for change based on #557

This will open up new possibilities for defining your own formatters.

### CODE REVIEW

**Comment by sulmar** [comment]
I just created a branch on which I started testing the export to Mermaid diagrams:
https://github.com/sulmar/stateless/blob/add-mermaid-graph-style

Still needs refinement but looks like it's starting to work after refactoring.

**Review by mclift — COMMENTED** [review]
Thanks, @sulmar

The new `abstract` method in `GraphStyleBase` will be a breaking change for existing derived classes, albeit a trivial one to fix. I've been considering whether adding it as a `virtual` method with a default implementation might be worth considering, but I don't imagine this change will affect many consumers of Stateless and creating a default implementation for backwards compatibility would just be creating technical debt and kicking the inevitable change further into the future. I'd be interested to know if others have any views on this.

File: src/Stateless/Graph/UmlDotGraphStyle.cs [hunk 1]
```diff
@@ -122,5 +123,21 @@ internal string FormatOneLine(string fromNodeName, string toNodeName, string lab
         {
             return $"\"{fromNodeName}\" -> \"{toNodeName}\" [style=\"solid\", label=\"{label}\"];";
         }
+
+        /// <summary>
+        ///
```
Comment (mclift) [inline src/Stateless/Graph/UmlDotGraphStyle.cs:128]: Missing summary (I think you could replace this with `/// <inheritdoc/>`)

**Comment by mclift** [comment]
3 days and no additional comments, so let's get this merged. Thanks, @sulmar

**Pull request merged**